app/src/components/common/SharingModal/ShareLinkView.tsx

```diff
@@ -129,11 +130,12 @@ export const ShareLinkView: React.FC<ShareLinkProps> = ({ selectedRules, source,
 
   const handleSharedListCreation = useCallback(() => {
     const isSharedListNameNotValid = validateSharedListName();
-    if (!sharedListName) {
+    if (sharedLinkVisibility === SharedLinkVisibility.PRIVATE && !sharedListName) {
       setError("Shared list name cannot be empty");
       return;
     }
-    if (isSharedListNameNotValid) {
+
+    if (sharedLinkVisibility === SharedLinkVisibility.PRIVATE && isSharedListNameNotValid) {
       setError("Shared list name cannot have special characters");
       return;
     }
```

Comment:
We can move both the `sharedlistName` and `isSharedListNameNotValid` checks inside a single `if(sharedLinkVisibility === SharedLinkVisibility.PRIVATE)`